src/custom/utils/operator.ts

```diff
@@ -203,23 +205,37 @@ function toApiAddress(address: string, chainId: ChainId): string {
   return address
 }
 
-async function _getJson(chainId: ChainId, url: string): Promise<any> {
-  let response: Response | undefined
+async function _getJson<T, E>(chainId: ChainId, url: string): Promise<ErrorHandlingResponse<T, E>> {
```

Comment:
I don't think we should model it like this. I feel is unnecessarily complex

Afaik, error 400s have a code, a code can tell us the type. 
So we can keep the plain promise. If there's an error we handle it. In the error handling we can check for the code, in case is a error code we identify

Reply:
you need to use the errorType if present, to see if it's one error you can handle

![image](https://user-images.githubusercontent.com/2352112/117953134-223ce280-b316-11eb-9070-847555dab1e4.png)